api/base/exceptions.py

```diff
@@ -1,5 +1,36 @@
-from rest_framework.exceptions import APIException
+
 from rest_framework import status
+from rest_framework.exceptions import APIException
+
+
+def json_api_exception_handler(exc, context):
+    """ Custom exception handler that returns errors object as an array """
+
+    # Import inside method to avoid errors when the OSF is loaded without Django
+    from rest_framework.views import exception_handler
+    response = exception_handler(exc, context)
+
+    # Error objects may have the following members. Title removed to avoid clash with node "title" errors.
+    top_level_error_keys = ['id', 'links', 'status', 'code', 'detail', 'source', 'meta']
+    errors = []
+
+    if response:
+        message = response.data
+        if isinstance(message, dict):
+            for key, value in message.iteritems():
+                if key in top_level_error_keys:
+                    errors.append({key: value})
+                else:
+                    errors.append({'detail': {key: value}})
```

Comment:
It's not exactly clear from the JSON API spec, but I don't think `detail` is meant to be a dict.

> detail: a human-readable explanation specific to this occurrence of the problem.

Are there places in the current API where we get to this branch?